### PR TITLE
chore: add tech sponsors

### PR DESCRIPTION
**Pull request by harish-sethuraman**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?
- add tech sponsors to readmes of all packages

#### Related Issues
fixes https://github.com/eslint/rewrite/issues/57

### CODE REVIEW

**Review by mdjermanovic — COMMENTED** [review]
This looks good like in eslint/eslint, but @nzakas mentioned "automatically" so I'm not sure if the goal was maybe to fetch data from https://raw.githubusercontent.com/eslint/eslint.org/main/src/_data/techsponsors.json instead of hardcoding.

**Comment by harish-sethuraman** [comment]
My bad I also get why its better to do it similar to sponsors and not have it statically hard coded. I have made the changes here. there are some open points. 

1. How do we want the layout to look? 
     - currently I have made it similar to gold sponsors. (as we were missing content related to the text. The texts in the JSON are very short)
3. Do we want to use the images from eslint.org site? or do we want to get it from somewhere else? 
     - added image from eslint.org for now

**Comment by nzakas** [comment]
We can have it look the same as the bronze sponsors, which is what we do on the website (https://eslint.org/sponsors).

File: tools/update-readme.js [hunk 1]
```diff
@@ -38,7 +44,7 @@ const heights = {
 
 /**
  * Fetches the latest sponsors data from the website.
- * @returns {Object} The sponsors data object.
+ * @returns {Object} The sponsors are data object.
```
Comment (nzakas) [inline tools/update-readme.js:47]: Mistake?

File: tools/update-readme.js [hunk 2]
```diff
@@ -74,12 +90,33 @@ function formatSponsors(sponsors) {
...
 }
 
+/**
+ * Formats an array of sponsors into HTML for the readme.
+ * @param {Array} sponsors The array of sponsors.
+ * @returns {string} The HTML for the readme.
+ */
+function formatTechSponsors(sponsors) {
+	return stripIndents`<!--techsponsorsstart-->
+        <h3>Tech Sponsors</h3>
```
Comment (nzakas) [inline tools/update-readme.js:100]:
```suggestion
        <h3>Technology Sponsors</h3>
```

File: tools/update-readme.js (hunk 1 above)
Comment (harish-sethuraman) [reply tools/update-readme.js:47]:
```suggestion
 * @returns {Object} The sponsors data object.
```

Comment (harish-sethuraman) [reply tools/update-readme.js]: Yea it was

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**